### ISSUE TITLE
Add handling of `collections.namedtuple` and `typing.NamedTuple`

### ISSUE THREAD

**Issue by nyggus** [nyggus/rounder#15]
Currently a `TypeError` is returned when `rounder` objects are fed in with instances of these two types:

```python
TypeError: __new__() missing 1 required positional argument: ...
```

Checking for both can be done as 

```python
isinstance(my_named_tuple, tuple) and hasattr(my_named_tuple, "_fields")
```

**Comment by nyggus** [comment]
This, added to `_do`, works perfectly fine:

```python
if isinstance(obj, tuple) and hasattr(obj, "_fields"):
    convert(obj._fields)
    return obj
```

**Comment by nyggus** [comment]
Actually, for the moment none of the two solutions work. The first does not round, and the second returns a tuple with fields.

**Comment by salabim** [comment]
It's all a bit more complicated. But this should work:

    def convert(obj):
        if isinstance(obj, complex):
            # this has to be checked before the Number check,
            # as complex is a Number
            return convert(obj.real) + convert(obj.imag) * 1j
        if isinstance(obj, numbers.Number):
            return func(obj, *digits)
        if isinstance(obj, list):
            obj[:] = list(map(convert, obj))
            return obj
        if isinstance(obj, tuple):
            if hasattr(obj, "_fields"): # it's a namedtuple
                return obj._replace(**convert(obj._asdict()))
            else:
                return tuple(convert(list(obj)))
        if isinstance(obj, set):
            return set(convert(list(obj)))
        if isinstance(obj, frozenset):
            return frozenset(convert(list(obj)))
        if isinstance(obj, (dict, collections.OrderedDict)):
            obj.update(zip(obj.keys(), convert(list(obj.values()))))
            return obj
        if isinstance(obj, array.array):
            obj[:] = array.array(obj.typecode, convert(obj.tolist()))
            return obj
        if hasattr(obj, "__dict__"):
            convert(obj.__dict__)
            return obj


        return obj

I had to add support for OrderedDict, which is now a feature!
Note that this also required moving the check for `__dict__` end of the function.

For clarity, I made the handling of tuple, set and frozenset more explicit.

**Comment by nyggus** [comment]
Implemented (`v0.3.2`).
